packages/react-native-gesture-handler/src/v3/HostGestureDetector.web.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect, useRef } from 'react';
+import { View } from 'react-native';
+import RNGestureHandlerModuleWeb from '../RNGestureHandlerModule.web';
+import { ActionType } from '../ActionType';
+export interface GestureHandlerDetectorProps {
+  handlerTags: number[];
+  dispatchesAnimatedEvents: boolean;
+  moduleId: number;
+  children?: React.ReactNode;
+}
+
+const HostGestureDetector = (props: GestureHandlerDetectorProps) => {
+  const { handlerTags, children } = props;
+
+  const viewRef = useRef(null);
+  const propsRef = useRef<GestureHandlerDetectorProps>(props);
+  const oldHandlerTags = useRef<Set<number>>(new Set<number>());
+
+  const attachHandlers = (): void => {
+    const currentHandlerTags = new Set(handlerTags);
+    const newHandlerTags: Set<number> = currentHandlerTags.difference(
+      oldHandlerTags.current
+    );
+    newHandlerTags.forEach((tag) => {
+      RNGestureHandlerModuleWeb.attachGestureHandler(
+        tag,
+        viewRef.current,
+        ActionType.NATIVE_DETECTOR,
```

Comment:
Shouldn't it be passed as prop? We also have `NATIVE_DETECTOR_ANIMATED_EVENT` which is not used anywhere (in a sense that it is not assigned to detector).